python/cudf/cudf/tests/test_orc.py

```diff
@@ -1009,6 +1009,180 @@ def test_pyspark_struct(datadir):
+        "lvl2_struct_map": po.Map(
+            key=po.String(),
+            value=po.Struct(**{"a": po.BigInt(), "b": po.BigInt()}),
+        ),
+    }
+
+    schema = po.Struct(**schema)
+
+    lvl1_map = [
+        rd.choice(
+            [
+                None,
+                [
+                    (
+                        rd.choice(al),
+                        rd.choice([None, np.random.randint(1, 1500)]),
+                    )
+                    for y in range(2)
+                ],
+            ]
+        )
+        for x in range(size)
+    ]
+    lvl2_map = [
+        rd.choice(
+            [
+                None,
+                [
+                    (
+                        rd.choice(al),
+                        rd.choice(
+                            [
+                                None,
+                                [
+                                    rd.choice(
+                                        [None, np.random.randint(1, 1500)]
+                                    )
+                                    for z in range(5)
+                                ],
+                            ]
+                        ),
+                    )
+                    for y in range(2)
+                ],
+            ]
+        )
+        for x in range(size)
+    ]
+    lvl2_struct_map = [
+        rd.choice(
+            [
+                None,
+                [
+                    (
+                        rd.choice(al),
+                        rd.choice(
+                            [
+                                None,
+                                (
+                                    rd.choice(
+                                        [None, np.random.randint(1, 1500)]
+                                    ),
+                                    rd.choice(
+                                        [None, np.random.randint(1, 1500)]
+                                    ),
+                                ),
+                            ]
+                        ),
+                    )
+                    for y in range(2)
+                ],
+            ]
+        )
+        for x in range(size)
+    ]
+
+    pdf = pd.DataFrame(
+        {
+            "lvl1_map": lvl1_map,
+            "lvl2_map": lvl2_map,
+            "lvl2_struct_map": lvl2_struct_map,
+        }
+    )
+    writer = po.Writer(
+        buff, schema, stripe_size=1024, compression=po.CompressionKind.NONE
+    )
+    tuples = list(
+        map(
+            lambda x: (None,) if x[0] is pd.NA else x,
+            list(pdf.itertuples(index=False, name=None)),
+        )
+    )
+
+    writer.writerows(tuples)
+    writer.close()
+
+    return buff
+
+
+map_buff = gen_map_buff(size=100000)
+
+
+@pytest.mark.parametrize(
+    "columns",
+    [None, ["lvl1_map", "lvl2_struct_map"], ["lvl2_struct_map", "lvl2_map"]],
+)
+@pytest.mark.parametrize("num_rows", [0, 15, 1005, 10561, 100000])
```

Comment:
100000 is enough to have multiple stripes?

Reply:
Yes, it has 98 stripes

Reply:
oh, it's writing with tiny stripes, now I see :D
carry on, then :)